### PR TITLE
Added functions to increase data usability.

### PR DESCRIPTION
**Pull request by robertburry**
Added two functions that can add some usability when getting a lot of METAR's. Also added an example of how a user might use this in a simple format, which can be expanded if the user wanted.

### CODE REVIEW

**Comment by robertburry** [comment]
Added importing Pandas library in lines 9-11.

Added "construct_urls" function in lines 108-134.

Added "pull_metars" function in lines 137-146.

Added example use case within a comment section in lines 180-193.

File: scripts/asos/iem_scraper_example.py [hunk 1]
```diff
@@ -102,6 +105,45 @@ def download_alldata():
...
+  try: 
+    startts = datetime.datetime(*start_date)
+    endts = datetime.datetime(*finish_date)
+
+    SERVICE = "http://mesonet.agron.iastate.edu/cgi-bin/request/asos.py?"
+    service = SERVICE + "data=all&tz=Etc/UTC&format=comma&latlon=yes&"
+
+    service += startts.strftime("year1=%Y&month1=%m&day1=%d&")
+    service += endts.strftime("year2=%Y&month2=%m&day2=%d&")
+    return [f"{service}&station={station}" for station in station]
```
Comment (akrherz) [inline scripts/asos/iem_scraper_example.py:131]: This is not correct when station is a list of strings or even when it is not.

File: scripts/asos/iem_scraper_example.py [hunk 2]
```diff
@@ -102,6 +105,45 @@ def download_alldata():
...
+  
+  except:
```
Comment (akrherz) [inline scripts/asos/iem_scraper_example.py:133]: This needs to be `except Exception:` to satiate LGTM

File: scripts/asos/iem_scraper_example.py [hunk 3]
```diff
@@ -6,6 +6,9 @@
 import time
 import datetime
 
+#allows user to import data into a pandas DataFrame.
+import pandas as pd
```
Comment (akrherz) [inline scripts/asos/iem_scraper_example.py:10]: I'm not necessarily keen on introducing a `pandas` requirement into a script that is meant to run on standard python

File: scripts/asos/iem_scraper_example.py [hunk 4]
```diff
@@ -134,3 +176,18 @@ def main():
...
     download_alldata()
     # main()
+
+    
+'''
+Example use case of using the construct_urls and pull_metar functions. 
+
+def main():
+    start = (2021,5,1)
+    finish = (2021,11,30)
```
Comment (akrherz) [inline scripts/asos/iem_scraper_example.py:186]: You wanna try hooking these up to a `sys.argv` parsing?

Comment (robertburry) [reply scripts/asos/iem_scraper_example.py:186]: I was going to use sys.argv, but found that using argparse was potentially a better usability for this kind of input.

**Comment by robertburry** [comment]
I ran a test case with the updated main using the added functions I purposed with the following command: 
`iem_scraper_example.py --start 2021 10 9 --finish 2021 10 10 >output.log` and the log file I got appeared to give a desired result of pulling the csv's that are hosted on IEM into memory using a list of tuples. 
[output.log](https://github.com/akrherz/iem/files/10021784/output.log)

I figured it was ok to leave the `#DEBUG` lines proceeding each csv as this does not adjust what you might download from IEM, but could be used as a parsing option if a user wanted to convert the list of tuples to a Pandas DataFrame or other format.

File: scripts/asos/iem_scraper_example.py [hunk 5]
```diff
@@ -134,3 +205,37 @@ def main():
 if __name__ == "__main__":
     download_alldata()
     # main()
+
+    
+'''
+Example use case of using the construct_urls and pull_metar functions. 
+
+def main():
```
Comment (akrherz) [inline scripts/asos/iem_scraper_example.py:213]: main() is already defined, so you'd want to rename this or remove the previous main()

File: scripts/asos/iem_scraper_example.py [hunk 6]
```diff
@@ -134,3 +205,37 @@ def main():
...
+    start = datetime.datetime(*args.start)
+    finish = datetime.datetime(*args.finish)
+
+    metar = []
+    # Can pull in from file, just needs to be List or Tuple.
+    station = ["CYOW", "CYYZ"]
+    
+    url = construct_urls(station, start, finish)
+
+    for i in url:
```
Comment (akrherz) [inline scripts/asos/iem_scraper_example.py:230]: `i` is usually an `int` and not a string url in this case.  Suggested change:
```
urls = construct_urls(station, start, finish)
for url in urls:
  metar += pull_metar(url)
```

This logic still eludes me as you would have a string with repeated headers.

File: scripts/asos/iem_scraper_example.py [hunk 7]
```diff
@@ -102,6 +109,70 @@ def download_alldata():
...
+            f'Start date {start_date} is past your finish_date {finish_date}')
+
+    try:
+
+        SERVICE = "http://mesonet.agron.iastate.edu/cgi-bin/request/asos.py?"
+        service = SERVICE + "data=all&tz=Etc/UTC&format=comma&latlon=yes&"
+
+        service += start_date.strftime("year1=%Y&month1=%m&day1=%d&")
+        service += finish_date.strftime("year2=%Y&month2=%m&day2=%d&")
+        return [f"{service}&station={station}" for station in station]
```
Comment (akrherz) [inline scripts/asos/iem_scraper_example.py:152]: A single URL call can have multiple `station` values set.  `station=XYZ&station=ABC&...`

**Comment by akrherz** [comment]
closing as stale, sorry.

**Pull request closed without merging**